### PR TITLE
 Fixed poll view when phone keypad is activated

### PR DESCRIPTION
**Pull request by MaianaB**
**Feature/Bug description:** Save survey when keypad is active.

**Solution:** Fixed poll view when phone keypad is activated

**TODO/FIXME:** n/a

captura de tela de 2018-12-06 14-07-35
captura de tela de 2018-12-06 14-08-12
captura de tela de 2018-12-06 14-08-35
captura de tela de 2018-12-06 14-08-41

### CODE REVIEW

File: frontend/survey/surveyDirective.js [hunk 1]
```diff
@@ -106,9 +106,24 @@
             return surveyCtrl.post.title && !notEnoughOptions && surveyCtrl.post.deadline;
         };
 
+        function defaultoptions(){
+            surveyCtrl.options = [];
+            surveyCtrl.options.push(angular.copy(option_empty));
```
Comment (JuliePessoa) [inline frontend/survey/surveyDirective.js]: All this function could just be this:
`surveyCtrl.options = [];`
`_.range(MIN_QUANTITY_OPTION).map(() => surveyCtrl.options.push(angular.copy(option_empty)))`

or
```
function reducer (acc) {
   acc.push(angular.copy(option_empty))
   return acc
}
survey.options = _.range(MIN_QUANTITY_OPTION).reduce(reducer, [])

File: frontend/survey/surveyDirective.js [hunk 2]
```diff
@@ -106,9 +106,24 @@
...
+                surveyCtrl.options.push(angular.copy(option_empty));
+                surveyCtrl.options.push(angular.copy(option_empty));
+            }
+        }
+
         function unobserveNewPost() {
             SubmitFormListenerService.unobserve("postCtrl.post");
         }
+
+        (function main() {
```
Comment (JuliePessoa) [inline frontend/survey/surveyDirective.js]: test

Comment (JuliePessoa) [reply frontend/survey/surveyDirective.js]: use $onInit and use change from directive to component

File: frontend/post/save_post_mobile.html [hunk 3]
```diff
@@ -62,6 +58,10 @@
           </md-button>
         </div>
       </form>
+      <!-- SURVEY -->
+      <survey-directive ng-if="postCtrl.typePost === 'Survey' && !isEditing" post="postCtrl.post" posts="posts" user="postCtrl.user" callback="postCtrl.choiceCommon"
```
Comment (JuliePessoa) [inline frontend/post/save_post_mobile.html]: theres no `posts` property anymore

File: frontend/survey/surveyDirective.js (hunk 1 above)
Comment (JuliePessoa) [reply frontend/survey/surveyDirective.js]: or just a `for` iterating over MIN_QUANTITY_OPTION

File: frontend/survey/save_survey_mobile.html [hunk 4]
```diff
@@ -19,29 +19,26 @@
...
+  </md-switch>
+  <md-button mdc-datetime-picker="" lang="pt-BR"
+  ng-model="surveyCtrl.post.deadline" class="md-fab md-primary save-survey-date-button"
+  today-text="Hoje" min-date="surveyCtrl.now"
+  readonly="readonly" cancel-text="Cancelar">
+    <md-icon>date_range</md-icon>
+  </md-button>
 </div>
+<md-card-actions layout="row" layout-align="end end"
+   class="save-survey-margin-buttons" style="bottom: 50px; padding-bottom: 8px;">
```
Comment (Luiz-FS) [inline frontend/survey/save_survey_mobile.html]: Move style to css file

File: frontend/survey/save_survey_mobile.html [hunk 5]
```diff
@@ -19,29 +19,26 @@
...
+  <md-button title="Cancelar" ng-click="surveyCtrl.resetSurvey()" class="save-survey-cancel-button">
+    CANCELAR
+  </md-button>
+  <md-button title="Enviar" ng-click="surveyCtrl.save()" style="margin-right: 0;"
```
Comment (Luiz-FS) [inline frontend/survey/save_survey_mobile.html]: Same.

File: frontend/survey/survey.component.js [hunk 6]
```diff
@@ -119,19 +130,15 @@
...
-                user: '=',
-                options: '=',
-                callback: '='
-            }
-        };
+    app.component("surveyComponent", {
+        templateUrl: getTemplateUrl(),
+        controllerAs: "surveyCtrl",
+        controller: "SurveyComponentController",
+        scope: {},
```
Comment (JuliePessoa) [inline frontend/survey/survey.component.js]: component does not have scope

**Pull request merged**